hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoTransferHandler.java

```diff
@@ -510,9 +510,13 @@ private void checkSender(
         // If the sender account is immutable, then we throw an exception.
         final var key = senderAccount.key();
         if (key == null || !isValid(key)) {
```

Comment:
This condition if `!isValid(` can be inside else for clarity

Reply:
With this suggestion do we expected the end version of the code to look like this? ->
```
final var key = senderAccount.key();
if (isHollow(senderAccount)) {
    meta.requireSignatureForHollowAccount(senderAccount);
} else {
    if (key == null || !isValid(key)) {
        // If the sender account has no key, then fail with INVALID_ACCOUNT_ID.
        // NOTE: should change to ACCOUNT_IS_IMMUTABLE
        throw new PreCheckException(INVALID_ACCOUNT_ID);
    } else if (!nftTransfer.isApproval()) {
        meta.requireKey(key);
    }
}

```

Reply:
yes Agreed @kselveliev

Reply:
Oh actually. I take it back. This makes sense also.